File: JLMailCore2Practice/JLMailCore2Practice/MainViewController.m
//
//  MainViewController.m
//  JLMailCore2Practice
//
//  Created by perfect on 2017/6/20.
//  Copyright © 2017年 JackLin. All rights reserved.
//

#import "MainViewController.h"
#import "IMAPViewController.h"
#import "POPViewController.h"
#import "SMTPViewController.h"
#import "SendMailViewController.h"
#import <MessageUI/MessageUI.h>

@interface MainViewController ()<UITableViewDelegate, UITableViewDataSource, MFMailComposeViewControllerDelegate>{
    NSArray *_functionsArray;
}

@property (nonatomic, strong) UITableView *mainTableView;

@end

@implementation MainViewController

#pragma mark - Life Cycle
- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"邮箱收发";
    self.view.backgroundColor = [UIColor whiteColor];
    _functionsArray = @[@"IMAP协议的使用", @"POP协议的使用", @"SMTP协议的使用"];
    [self.view addSubview:self.mainTableView];
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - Getters
- (UITableView *)mainTableView{
    if (!_mainTableView) {
        CGRect rect = CGRectMake(0, 0, CGRectGetWidth(self.view.frame), CGRectGetHeight(self.view.frame));
        _mainTableView = [[UITableView alloc] initWithFrame:rect];
        _mainTableView.delegate = self;
        _mainTableView.dataSource = self;
        _mainTableView.tableFooterView = [UIView new];
    }
    return _mainTableView;
}

#pragma mark - Action
#pragma mark - Private
#pragma mark - UITableViewDataSource
- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView{
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return _functionsArray.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    static NSString *cellId = @"mailFuncTableCell";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:cellId];
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:cellId];
    }
    cell.textLabel.text = _functionsArray[indexPath.row];
    return cell;
}

#pragma mark - UITableViewDelegate
- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath{
    return 75;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
    
    switch (indexPath.row) {
        case 0:{
            
            IMAPViewController *imapVC = [[IMAPViewController alloc] init];
            [self.navigationController pushViewController:imapVC animated:YES];
            
        }
          break;
        case 1:{
            
            POPViewController *popVC = [[POPViewController alloc] init];
            [self.navigationController pushViewController:popVC animated:YES];
            
        }
            break;
        case 2:{
            /*SMTPViewController *smtpVC = [[SMTPViewController alloc] init];
            [self.navigationController pushViewController:smtpVC animated:YES];*/
            //SendMailViewController *sendVC = [[SendMailViewController alloc] init];
            //[self.navigationController pushViewController:sendVC animated:YES];
            
            if (MFMailComposeViewController.canSendMail) {
                MFMailComposeViewController *mailVC = [[MFMailComposeViewController alloc] init];
                mailVC.mailComposeDelegate = self;
                [mailVC setSubject: @"eMail主题"];
                NSArray *toRecipients = [NSArray arrayWithObject: @"first@example.com"];
                [mailVC setToRecipients:toRecipients];
                [self presentViewController:mailVC animated:YES completion:nil];
            }
            
            
        }
            break;
            
        default:
            break;
    }
}


- (void)mailComposeController:(MFMailComposeViewController *)controller didFinishWithResult:(MFMailComposeResult)result error:(nullable NSError *)error{
    
}


@end

File: JLSwiftPractice/JLSwiftPractice/MainViewController.swift
//
//  MainViewController.swift
//  JLSwiftPractice
//
//  Created by jacklin on 2017/9/17.
//  Copyright © 2017年 jacklin. All rights reserved.
//

import UIKit
import SnapKit
import Alamofire
import AlamofireObjectMapper
import SwiftyJSON
import ObjectMapper

let kBaseUrl = "https://is.snssdk.com/"
let IID: String = "5034850950"
let device_id: String = "6096495334"

class MainViewController: UIViewController {

    let dataSource = ["Alamofire", "snapKit", "基本框架搭建"]
    
    lazy fileprivate var mainTableView : UITableView = {
        let tableView = UITableView()
        tableView.backgroundColor = UIColor.clear
        tableView.delegate = self;
        tableView.dataSource = self;
        tableView.tableFooterView = UIView()
        return tableView
    }()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        title = "主页"
        view.backgroundColor = UIColor.white
        
        view.addSubview(mainTableView)
        mainTableView.snp.makeConstraints { (make) in
            make.edges.equalTo(self.view)
        }
        
        //演示一些第三库的基本使用方法
        //整个框架的组织结构、初始化一些基类
        //封装一些网络框架

    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}

extension MainViewController: UITableViewDataSource {
    
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1;
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return dataSource.count;
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cellIdentifer = "tableViewCell"
        var cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifer)
        if cell == nil {
            cell = UITableViewCell(style: .default, reuseIdentifier: "tableViewCell")
        }
        
        cell?.textLabel?.text = dataSource[indexPath.row]
        
        return cell!
    }
}

extension MainViewController: UITableViewDelegate{
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        
        if indexPath.row == 0 {
            //测试Alamofire使用方法
            
            let requestUrl = kBaseUrl + "article/category/get_subscribed/v1/?"
            let parameters = [device_id: device_id,
                              "aid": 13 ,
                              "iid": IID] as [String : Any]
            Alamofire.request(requestUrl, parameters: parameters).responseJSON { (reponse) in
                
                guard reponse.result.isSuccess else{
                    return
                }
                
                if let value = reponse.result.value{
                    
                    let json = JSON(value)
                    
                    let dataDic = json.dictionary
                    
                    let newsModel = Mapper<NewsDataModel>().map(JSON: value as! [String : Any])
                    
                    print("string = \(newsModel?.toJSONString())")
                    
                    /*let dataNewModel = Mapper<NewsDataModel>().map(dataDic)
                    
                    let dataModel = NewsDataModel()
                    _ = Mapper().map(JSON: dataDic!, toObject: dataModel)
                    
                    print("data = version : \(dataModel)")*/
                    
                    
                    
                }
                
                
                
                
            }
            
            
            
            
            
        }
        
    }
    
}
